src/commands/service/pipeline.ts

```diff
@@ -62,6 +65,29 @@ export const execute = async (
 ) => {
   try {
     await fetchValues(serviceName, opts);
+    const accessOpts: IAzureDevOpsOpts = {
+      orgName: opts.orgName,
+      personalAccessToken: opts.personalAccessToken,
+      project: opts.devopsProject
+    };
+
+    // By default the version descriptor is for the master branch
+    const hasPipelineFile = await repositoryHasFile(
+      SERVICE_PIPELINE_FILENAME,
+      opts.yamlFileBranch ? opts.yamlFileBranch : "master",
+      opts.repoName,
+      accessOpts
+    );
+
+    if (!hasPipelineFile) {
+      logger.error(
+        "Error installing build pipeline. Repository does not have a " +
+          SERVICE_PIPELINE_FILENAME +
+          " file."
+      );
+      await exitFn(1);
+    }
+
```

Comment:
same here, `else` is needed. 
Can you please create a helper function for this hasPipelineFile check because the same code is copied 3 times?

Reply:
Sure, I can put all the logic in azdoClient instead of only the piece that checks if the file is present.

Reply:
maybe `azdoutil.ts` is ok too

Reply:
I would suggest using `azdoClient.ts` since we are planning to remove [`azdoutil.ts`](https://github.com/CatalystCode/spk/blob/4e48b6a28f9b64d196b41ea7358c5e17d4c2b832/src/lib/azdoutil.ts#L6) which has one duplicate function that also exist in [`azdoClient.ts`](https://github.com/CatalystCode/spk/blob/4e48b6a28f9b64d196b41ea7358c5e17d4c2b832/src/lib/azdoClient.ts#L19).

Reply:
Yeah, there's very little in azdoutil, and I already have shared functionality in azdoClient as part of this pr